src-goocanvas2/gooCanvas.ml

```diff
@@ -0,0 +1,228 @@
+(**************************************************************************)
+(*                Lablgtk                                                 *)
+(*                                                                        *)
+(*    This program is free software; you can redistribute it              *)
+(*    and/or modify it under the terms of the GNU Library General         *)
+(*    Public License as published by the Free Software Foundation         *)
+(*    version 2, with the exception described in file COPYING which       *)
+(*    comes with the library.                                             *)
+(*                                                                        *)
+(*    This program is distributed in the hope that it will be useful,     *)
+(*    but WITHOUT ANY WARRANTY; without even the implied warranty of      *)
+(*    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the       *)
+(*    GNU Library General Public License for more details.                *)
+(*                                                                        *)
+(*    You should have received a copy of the GNU Library General          *)
+(*    Public License along with this program; if not, write to the        *)
+(*    Free Software Foundation, Inc., 59 Temple Place, Suite 330,         *)
+(*    Boston, MA 02111-1307  USA                                          *)
+(*                                                                        *)
+(*                                                                        *)
+(**************************************************************************)
+
+open Gaux
+open GtkGooCanvas
+open GooCanvasEnums
+open Gobject
+open Gtk
+open GtkBase
+open GooCanvas_types
+open OgtkGooCanvasProps
+open GObj
+
+let coord_array_to_tuple = function
+| [| x1 ; y1 ; x2 ; y2 |] -> (x1, y1, x2, y2)
+| _ -> assert false
+
+module Tags = struct
+  include (GooCanvasEnums : module type of GooCanvasEnums
+                      with module Conv := GooCanvasEnums.Conv)
+end
+
+let font_desc_conv = (Gobject.Data.unsafe_pointer : Pango.font_description data_conv)
+let canvas_points_conv = (Gobject.Data.unsafe_pointer : GtkGooCanvas.canvas_points data_conv)
+let pointer_events_conv = Gobject.Data.flags GooCanvasEnums.Conv.pointer_events_tbl
+
+let propertize = function
```

Comment:
Why is the conversion done by hand rather than use `propcc` ?

Reply:
I took the code of GnoCanvas as example back in 2018 when I started the bindings to goocanvas2, then I did not follow all the changes which occured in lablgtk3. I commited a change to use generated conversion functions, except for canvas-points: I don't know the right way to use propcc to handle this structure https://developer.gnome.org/goocanvas/stable/goocanvas-GooCanvas-Types.html#GooCanvasPoints .

Reply:
Let me know if there is a more idiomatic way to replace the "propertize" function.